### PR TITLE
upgrade rke, helm, and kubernetes versions 

### PR DESCRIPTION
**Pull request by steved**
The new CIS values come from the [2.4 guide](https://rancher.com/docs/rancher/v2.x/en/security/rancher-2.4/hardening-2.4/).

### CODE REVIEW

File: ansible/roles/rke/vars/main.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
-kubectl_version: v1.15.2
-helm_version: v2.16.1
-rancher_cli_version: v2.2.0
+kubectl_version: v1.17.12
+helm_version: v3.2.1
```
Comment (sonnysideup) [inline ansible/roles/rke/vars/main.yml:3]: Any reason not to use the latest version here? (v1.19.0 and v3.3.4, respectively)

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ commands:
       ssh_user:
         type: string
     steps:
-    - run: '[[ $CIRCLE_BRANCH != "master" && -z $CIRCLE_PULL_REQUEST ]] && circleci step halt || true'
+    # - run: '[[ $CIRCLE_BRANCH != "master" && -z $CIRCLE_PULL_REQUEST ]] && circleci step halt || true'
```
Comment (sonnysideup) [inline .circleci/config.yml:38]: :scissors: ?

File: ansible/roles/rancher/tasks/rancher.yml [hunk 3]
```diff
@@ -5,10 +5,9 @@
 
 - name: Install Rancher
   command: >
-    helm install rancher-stable/rancher
-      --name rancher
+    helm install rancher rancher-stable/rancher
       --namespace cattle-system
-      --version {{ rancher_version | default('2.3.5', true) }}
+      --version {{ rancher_version | default('2.3.6', true) }}
```
Comment (sonnysideup) [inline ansible/roles/rancher/tasks/rancher.yml:10]: Should this default be here or in a vars file? I know you didn't add this here originally but it seems a litte out of place.

**Review by sonnysideup — COMMENTED** [review]
Ah yes, I remember these good ol' rancher days. Seems fine and config changes correspond with the 2.4 hardening guide. Check out my comments.

File: ansible/roles/rancher/tasks/rancher.yml (hunk 3 above)
Comment (steved) [reply ansible/roles/rancher/tasks/rancher.yml]: Nope, in the var file makes more sense :+1:

File: ansible/roles/rke/vars/main.yml (hunk 1 above)
Comment (steved) [reply ansible/roles/rke/vars/main.yml]: not a good one, fixed

**Pull request merged**